src/corefx/System.Globalization.Native/pal_locale.c

```diff
@@ -156,7 +156,14 @@ const char* DetectDefaultLocaleName()
         }
     }
```

Comment:
Can we delete the `if (loc != NULL && (strcmp("C", loc) == 0 || strcmp("POSIX", loc) == 0))` part as well? I think it should be covered by the code below.

Reply:
I was hoping anyone doing 

```
new CultureInfo("C")
```

will get invariant for the same reason we have changed it in the default locale. otherwise, I am pretty sure many people will run into the exact same problem in some scenarios (when getting the locale name from other sources e.g. HTML, XML, JSON files)

Reply:
will look at this and know the behavior when we have this case.

Reply:
I tried that and you are right. ICU keep mapping to en_US_POSIX when LANG and LC_ALL are missing. I'll remove this part too.

Reply:
Thanks @jkotas, I think the code is cleaner and simple to explain our new behavior.